### PR TITLE
Tighten visual design of charts

### PR DESCRIPTION
**Pull request by kdoh**
A plethora of small tweaks to crisp up our charts and make dashboards nicer to look at. The commit log is a pretty good overview of what's here.

- remove box-shadow from dashboard cards, add border
- set scalar font weight to 700
- remove extra bold style from dash headings
- set font weight for row and line axis ticks and labels to 700, adjust size
- tweak gridline dash array size for cleaner lines
- make scalar value titles the same as other dashcard titles
- even out legend title spacing on cards
- lighten axis domain line
- use border color for axis line on RowChart
- make legend 2nd read, adjust spacing & alignment
- adjust dashcard legend size for non linebar viz

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Create a new dashboard manually with a variety of chart types, but specifically scalars, trends, and line area bar charts. An easy way to do this is to create a dash via X-ray off of the sample dataset.
2. The changes listed above should be visible.

### Checklist

 - [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/LineAreaBarChart.css [hunk 1]
```diff
@@ -7,8 +7,9 @@
 }
 
 .LineAreaBarChart .dc-chart .grid-line.horizontal {
-  stroke: color-mod(var(--color-text-medium) alpha(-80%));
-  stroke-dasharray: 5, 5;
+  stroke: var(--color-text-medium) !important;
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/LineAreaBarChart.css:10]: I think something about our css load order was causing these to just use the default dc.js styling.

File: frontend/src/metabase/visualizations/components/legend/LegendItem.styled.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const LegendItemRoot = styled.div<{ isVertical: boolean }>`
 
 export const LegendItemLabel = styled.div<{ isMuted: boolean }>`
   display: flex;
-  align-items: baseline;
+  align-items: center;
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/legend/LegendItem.styled.tsx:19]: Wasn't sure if there was a good reason these were `baseline` but it was causing some weird alignment issues.
Screenshot 2023-12-20 at 11 06 00 AM

**Review by alxnddr — APPROVED** [review]
Looks good, I'll reflect these changes in the ECharts branch

File: frontend/src/metabase/dashboard/components/DashboardGrid.styled.tsx [hunk 3]
```diff
@@ -31,7 +32,7 @@ export const DashboardCard = styled.div<DashboardCardProps>`
     bottom: 0;
     right: 0;
     border-radius: 8px;
-    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.13);
+    border: 1px solid ${colors["border"]};
```
Comment (kulyk) [inline frontend/src/metabase/dashboard/components/DashboardGrid.styled.tsx:35]: I'm not sure if `colors` works with whitelabeling. We should be safe if we use `color` instead:

```ts
import { color } from "metabase/lib/colors";

border: 1px solid ${color("border")};
```

**Review by kulyk — APPROVED** [review]
LGTM!

**Pull request merged**